### PR TITLE
DOC: Use references bibliography file for DIPY citation file

### PR DESCRIPTION
**Pull request by jhlegarreta**
Use references bibliography file for DIPY citation file.

Add missing citations to `references.bib`.

Add a missing comma after the `Garyfallidis2014` cite key (now renamed to `Garyfallidis2014` since another work from the same author/year is added in this commit).

Remove the `:list: enumerated` flag in `bibliography.rst` to avoid
```
/home/runner/work/dipy/dipy/doc/cite.rst:5:
 WARNING: could not find bibtex key "Garyfallidis2014a"
```

and similar errors.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Increases the number of warnings so making it a draft for now.

**Comment by jhlegarreta** [comment]
These should also be transitioned:
https://docs.dipy.org/stable/reconstruction_models_list.html

**Review by skoudoro — COMMENTED** [review]
Hi @jhlegarreta,

Overall, it looks good. 

I just have a doubt about the `README.rst`. because it will look strange on DIPY Github homepage.  For sure it will not be processed and we do not use the `README` in the doc generation.

I think you should revert these changes on the `README`. 

Thank you for the future update

**Comment by jhlegarreta** [comment]
:+1: Removed with the last force push.

**Comment by jhlegarreta** [comment]
The doc build is generating warnings in the code base:
https://github.com/dipy/dipy/actions/runs/10586647103/job/29335853394?pr=3319#step:5:1750

Compare to:
https://github.com/dipy/dipy/actions/runs/10569986141/job/29283731136#step:5:1669

e.g.
https://github.com/dipy/dipy/actions/runs/10586647103/job/29335853394?pr=3319#step:5:893

That reference does exist in this commit:
https://github.com/dipy/dipy/blob/3baae5048438959c96f315594e0d391f9504a021/doc/references.bib#L2057

And the refs in the rst file are not being found:
https://github.com/dipy/dipy/actions/runs/10586647103/job/29335853394?pr=3319#step:5:1084

So not sure what happens here. Cannot devote time to this now, so @skoudoro let me know when you have time to have a look. Thanks.

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @jhlegarreta,

Thank you for this.

for some reason, it works as expected when we remove `:list: enumerated` from https://github.com/dipy/dipy/blob/ec07b657b22f7baca0eecd9777fa9a918fea12fb/doc/user_guide/bibliography.rst?plain=1#L8

So I proposed that you remove it for now.

ref: https://github.com/jupyter-book/jupyter-book/issues/1662

See also my comment below. 

Last point, it does not look so great (even if the references works well). We need a better way to render this.

image

File: doc/references.bib [hunk 1]
```diff
@@ -634,6 +634,16 @@ @article{Garyfallidis2015
   url       = {https://doi.org/10.1016/j.neuroimage.2015.05.016}
 }
 
+@article{Garyfallidis2014,
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply doc/references.bib]: :ok_hand:. Fixed in 0a14e04. Force pushed. Also, maybe the absence of a comma after the cite key explained the missing citations. I am waiting for the CI to remove the `:list: enumerated` option.

**Comment by jhlegarreta** [comment]
> Last point, it does not look so great (even if the references works well). We need a better way to render this.

Using biblatex there is a way to put the entire citation at a given place instead of using a cross-reference using `fullcite`. That would be ideal here, but did not find a way in `sphinxbibtex-contrib`.

Honestly, I would prioritize here consistency and reduce maintenance burden, which I would say we achieve with these changes. People can click on the links and C&P the refs or copy the bibtex, the later not being possible without these changes.

File: doc/references.bib (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply doc/references.bib]: > I am waiting for the CI result to remove the :list: enumerated option.

Errors persisted with the option, so removed in the last force push.

**Review by skoudoro — APPROVED** [review]
ok, all good, Thanks @jhlegarreta, merging

**Pull request merged**